L1Trigger/L1TZDC/plugins/L1TZDCAnalyzer.cc

```diff
@@ -0,0 +1,154 @@
+
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+
+#include "CondFormats/L1TObjects/interface/CaloParams.h"
+#include "CondFormats/DataRecord/interface/L1TCaloParamsRcd.h"
+
+#include "DataFormats/L1TCalorimeter/interface/CaloTower.h"
+#include "DataFormats/L1TCalorimeter/interface/CaloCluster.h"
+#include "DataFormats/L1Trigger/interface/EGamma.h"
+#include "DataFormats/L1Trigger/interface/Tau.h"
+#include "DataFormats/L1Trigger/interface/Jet.h"
+#include "DataFormats/L1Trigger/interface/EtSum.h"
+
+//For the output
+#include "TTree.h"
+//string for some branch handling
+#include <string>
+
+//
+// class declaration
+//
+
+namespace l1t {
+
+  class L1TZDCAnalyzer : public edm::one::EDAnalyzer<edm::one::SharedResources> {
+  public:
+    explicit L1TZDCAnalyzer(const edm::ParameterSet&);
+    ~L1TZDCAnalyzer() override;
+
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  private:
+    void beginJob() override;
+    void analyze(const edm::Event&, const edm::EventSetup&) override;
+    void endJob() override;
+
+    //fileservice
+    edm::Service<TFileService> m_fs;
+    //Declare a tree, member and pointer
+    TTree* m_zdcEtSumTree_p;
+    //Declare the etSum max and bpx max
+    //MAX Here is effectively a hardcoded number - you just need to know it from the producer - realistically this is best handled by something passed but for now (2023.08.17) we hardcode
+    static const int m_maxBPX = 5;
+    float m_zdcEtSumP[m_maxBPX];
+    float m_zdcEtSumM[m_maxBPX];
+
+    //virtual void beginRun(edm::Run const&, edm::EventSetup const&) override;
+    //virtual void endRun(edm::Run const&, edm::EventSetup const&) override;
+    //virtual void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+    //virtual void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+
+    // ----------member data ---------------------------
+    edm::EDGetToken m_sumToken;
+
+    bool doText_;
+    bool doHistos_;
+
+    TFileDirectory evtDispDir_;
+  };
+
+  //
+  // constants, enums and typedefs
+  //
+
+  //
+  // static data member definitions
+  //
+
+  //
+  // constructors and destructor
+  //
+  L1TZDCAnalyzer::L1TZDCAnalyzer(const edm::ParameterSet& iConfig)
+      : doText_(iConfig.getUntrackedParameter<bool>("doText", true)),
+        doHistos_(iConfig.getUntrackedParameter<bool>("doHistos", true)) {
+    usesResource(TFileService::kSharedResource);
+    //now do what ever initialization is needed
+
+    // register what you consume and keep token for later access:
+    edm::InputTag nullTag("None");
+
+    edm::InputTag sumTag = iConfig.getParameter<edm::InputTag>("etSumToken");
+    m_sumToken = consumes<l1t::EtSumBxCollection>(sumTag);
+
+    std::cout << "Processing " << sumTag.label() << std::endl;
+  }
+
+  L1TZDCAnalyzer::~L1TZDCAnalyzer() {
```

Comment:
Remove and declare it `= default;` above.

Reply:
Done in latest commit